internal/api/json.go

```diff
@@ -0,0 +1,72 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/shulganew/shear.git/internal/appconsts"
+	"github.com/shulganew/shear.git/internal/config"
+	"github.com/shulganew/shear.git/internal/service"
+	"go.uber.org/zap"
+)
+
+type RequestJSON struct {
+	LongURL string `json:"url"`
+}
+
+type ResonseJSON struct {
+	ShortURL string `json:"result"`
+}
+
+type HandlerAPI struct {
+	serviceURL *service.Shortener
+	conf       *config.Shear
+}
+
+func (u *HandlerAPI) GetService() service.Shortener {
+	return *u.serviceURL
+}
+
+func (u *HandlerAPI) SetAPI(res http.ResponseWriter, req *http.Request) {
```

Comment:
а почему такое название выбрано? Кажется так будет лучше:
```suggestion
func (u *HandlerAPI) GetShortURL(res http.ResponseWriter, req *http.Request) {
```

Reply:
Я так понял, что это REST и это устанавливаемое значение в API, согласен, так будет логичнее.